src/rgw/rgw_auth.h

```diff
@@ -602,6 +610,12 @@ class LocalApplier : public IdentityApplier {
   void load_acct_info(const DoutPrefixProvider* dpp, RGWUserInfo& user_info) const override; /* out */
   uint32_t get_identity_type() const override { return TYPE_RGW; }
   string get_acct_name() const override { return {}; }
+  string get_user_id() const override {
+    if (subuser != NO_SUBUSER) {
+      return user_info.user_id.id + ":" + subuser;
```

Comment:
this shouldn't require the other side to do string parsing to get the `:subuser` part. the user_id part is duplicated in user_info, so better to just add a `get_subuser()` that either returns `optional<string>` or an empty string when there isn't one

Reply:
Makes sense. Done.